src/analysis/retail/shaman/elemental/modules/features/SpenderWindow.tsx

```diff
@@ -0,0 +1,295 @@
+import ThresholdPerformancePercentage, {
+  LTEThreshold,
+  GTEThreshold,
+} from './shared/ThresholdPerformancePercentage';
+import SPELLS from 'common/SPELLS';
+import TALENTS from 'common/TALENTS/shaman';
+import { Talent } from 'common/TALENTS/types';
+import { formatDuration } from 'common/format';
+import { SpellIcon, SpellLink, TooltipElement } from 'interface';
+import { explanationAndDataSubsection } from 'interface/guide/components/ExplanationRow';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { CastEvent } from 'parser/core/Events';
+import Enemies from 'parser/shared/modules/Enemies';
+import { ON_CAST_BUFF_REMOVAL_GRACE_MS } from '../../constants';
+
+interface ActiveSpenderWindow {
+  timestamp: number;
+  motePresent: boolean;
+  flameshockPresent: boolean;
+}
+
+interface FinishedSpenderWindow extends ActiveSpenderWindow {
+  elshocksPresent: boolean;
+  sopUse: CastEvent;
+}
+
+const SOP_SPENDERS = [
```

Comment:
[Nit] I got a little confused that there are two different types of spenders in this file, maelstrom spenders and SOP spenders.  You might consider changing the the name of one.  At a glance `onSpender` / `onSopSpender` and `sopSpender` / `stSpender` look pretty similar.

Reply:
Renamed MS related spenders to include MS in the name, and renamed SoP spenders to SoP consumers.